### PR TITLE
ar71xx: add support for TL-WR802N Version 2

### PR DESCRIPTION
**Pull request by TPRoberts**
Signed-off-by: Thomas Roberts <tom.p.roberts@gmail.com>

### CODE REVIEW

File: target/linux/ar71xx/base-files/etc/board.d/02_network [hunk 1]
```diff
@@ -113,6 +113,7 @@ ar71xx_setup_interfaces()
 	tl-wa901nd-v4|\
 	tl-wr703n|\
 	tl-wr802n-v1|\
+    tl-wr802n-v2|\
 	tube2h|\
```
Comment (blogic) [inline target/linux/ar71xx/base-files/etc/board.d/02_network]: there is a whitespace error here. please make sure to use tabs and not space.

**Comment by blogic** [comment]
looking good, please fix the 2 whitespace errors

**Comment by TPRoberts** [comment]
Thanks , all corrected.

File: target/linux/ar71xx/files/arch/mips/ath79/mach-tl-wr802n-v2.c [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	NULL,
+};
+
+static struct flash_platform_data tl_wr802n_v2_flash_data = {
+	.part_probes	= tl_wr802n_v2_part_probes,
+};
+
+static struct gpio_led tl_wr802n_v2_leds_gpio[] __initdata = {
+	{
+		.name		= "tp-link:green:system",
```
Comment (heinzek) [inline target/linux/ar71xx/files/arch/mips/ath79/mach-tl-wr802n-v2.c]: Change led name accordance with the standard. Like board:color:funcion 
tl-wr802n-v2:green:system

**Comment by bobafetthotmail** [comment]
Please squash the commits into one.

In your local git branch of this PR write **git reset HEAD~2** to delete the commits leaving files intact.
**git add -i** and use its interface to add again all files you added/modified
**git commit**  and you write the commit title and message
**git push -f** to send the updated commit to Github and in this PR.

**Comment by TPRoberts** [comment]
Corrected and merged into one commit

File: target/linux/ar71xx/base-files/etc/board.d/01_leds [hunk 3]
```diff
@@ -610,6 +610,9 @@ tl-wpa8630)
 tl-wr802n-v1)
 	ucidef_set_led_wlan "wlan" "WLAN" "tp-link:blue:system" "phy0tpt"
 	;;
+tl-wr802n-v2)
+	ucidef_set_led_wlan "wlan" "WLAN" "tp-link:green:system" "phy0tpt"
```
Comment (heinzek) [inline target/linux/ar71xx/base-files/etc/board.d/01_leds]: When change led name You need to change this to "$board:green:system"

File: target/linux/ar71xx/base-files/etc/diag.sh [hunk 4]
```diff
@@ -375,6 +375,7 @@ get_status_led() {
 	tl-wr710n|\
 	tl-wr720n-v3|\
 	tl-wr802n-v1|\
+	tl-wr802n-v2|\
```
Comment (heinzek) [inline target/linux/ar71xx/base-files/etc/diag.sh]: move tl-wr802n-v2|\ to section with "$board:green:system"

File: target/linux/ar71xx/image/tp-link.mk [hunk 5]
```diff
@@ -599,6 +599,16 @@ define Device/tl-wr802n-v1
 endef
 TARGET_DEVICES += tl-wr802n-v1
 
+define Device/tl-wr802n-v2
+    $(Device/tplink-4mlzma)
+    DEVICE_TITLE := TP-LINK TL-WR802N v2
+	BOARDNAME := TL-WR802N-v2
```
Comment (heinzek) [inline target/linux/ar71xx/image/tp-link.mk]: Here use spaces

**Comment by TPRoberts** [comment]
Corrected.

**Comment by mkresin** [comment]
There is still a merge commit in your PR which should not be there.

Please have a look at https://github.com/bobafetthotmail/source/blob/c7cfdff4fa40f53511b8c8a4bd2e60aa980dc6fd/.github/pull_request_template.

One of the commits need a proper subject, the other one does not have a Signed-off-by line.

Please add some informations about the board to the commit message. Have a look at https://github.com/lede-project/source/commit/3f31029b190417e12f4d19d9999cd52ec76d0f85 for a good example how it could look like.

**Comment by TPRoberts** [comment]
Thanks, updated the commit.

**Comment by blogic** [comment]
merged, Thanks !

**Pull request closed without merging**